packages/flutter/lib/src/material/page.dart

```diff
@@ -118,7 +132,8 @@ mixin MaterialRouteTransitionMixin<T> on PageRoute<T> {
   @override
   Widget buildTransitions(BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation, Widget child) {
     final PageTransitionsTheme theme = Theme.of(context).pageTransitionsTheme;
-    return theme.buildTransitions<T>(this, context, animation, secondaryAnimation, child);
+    delegatedTransition = theme.delegatedTransition(context);
```

Comment:
I commented on this above in the example after discussing in person. The format of how the delegatedTransition will have to change. Ideally it will be provided alongside the transition itself, maybe part of the PageTransitionsBuilder.